migrate/migrate.go

```diff
@@ -87,6 +87,48 @@ func List(ctx context.Context, session gocqlx.Session) ([]*Info, error) {
 	return v, nil
 }
 
+// Pending provides a listing of pending migrations.
+func Pending(ctx context.Context, session gocqlx.Session, f fs.FS) ([]*Info, error) {
+	applied, err := List(ctx, session)
+	if err != nil {
+		return nil, err
+	}
+
+	fm, err := fs.Glob(f, "*.cql")
+	if err != nil {
+		return nil, fmt.Errorf("list migrations: %w", err)
+	}
+
+	if len(applied) > len(fm) {
+		return nil, fmt.Errorf("database is ahead")
```

Comment:
no need to have validation here, if it is called asynchronously you will get into troubles.
If you want to validate it, just add another method `ValidateFS(ctx context.Context, session gocqlx.Session, f fs.FS) error` and do just that there.